daffodil-test/src/test/scala/org/apache/daffodil/layers/TestLayers.scala

```diff
@@ -44,4 +45,9 @@ class TestLayers {
   @Test def test_layersErr1() { runner.runOneTest("layersErr1") }
   @Test def test_layers4() { runner.runOneTest("layers4") }
 
+  @Test def test_midBitsToEndMSBF1() { midBitsToEndRunner.runOneTest("midBitsToEndMSBF1") }
+  @Test def test_midBitsToEndLSBF1() { midBitsToEndRunner.runOneTest("midBitsToEndLSBF1") }
+
+  @Test def test_midBitsToEnd_unsufficentBits_1() { midBitsToEndRunner.runOneTest("midBitsToEnd_unsufficentBits_1") }
+
```

Comment:
Would it be possible to add more tests? There's a lot of complex bit logic/shifting going on up above. Would be nice to see some variation in tests to try to make sure edge cases are covered, e.g. firstBitIndex and/or numBits more than 8 bits, invalid argument values, etc.